src/transformers/models/conditional_detr/modeling_conditional_detr.py

```diff
@@ -1731,7 +1731,7 @@ def forward(
         # class logits + predicted bounding boxes
         logits = self.class_labels_classifier(sequence_output)
 
-        reference = outputs.reference_points if return_dict else outputs[-1]
+        reference = outputs.reference_points if return_dict else outputs[-2]
```

Comment:
from 

```
        if not return_dict:
            return decoder_outputs + encoder_outputs
```
I think the correct index really depends on `output_attentions` and `output_hidden_states`  ...?

Reply:
Yes, you are right, it's actually hard to determine the correct index, and it may introduce many if/else. For now, it was broken, but no one reported so I assume it's this kind of fix is fine. Don't think `return_dict=False` usage exists for this model actually 😄

Reply:
Yeah (I hate `return_dict=False`!)
But maybe add a comment around here (so anyone in the future might save some hours) if something happen to them

Reply:
Sounds good 👍